pandas/io/excel.py

```diff
@@ -615,13 +686,19 @@ def _parse_cell(cell_contents, cell_typ):
                 row = []
                 for j, (value, typ) in enumerate(zip(sheet.row_values(i),
                                                      sheet.row_types(i))):
-                    if usecols is not None and j not in should_parse:
-                        should_parse[j] = self._should_parse(j, usecols)
+                    if (j not in should_parse) and (_is_int_or_intlist(usecols)
```

Comment:
rather than checking with a bespoke function, this can just fall thru to the TextParser which validates all of this

Reply:
Because `usecols` can not be an int in TextParser and because `index_col` is handled differently in TextParser so that.

In `read_excel` the keyword argument `index_col` make index of the `DataFrame` the _mth_ column of the _filtered columns_ while in TextParser the `index_col` make as an index of the `DataFrame` the _mth_ column of the `DataFrame` whether is in the selected columns in `usecols` or not. Do you understand why I did this?

Then, after check the cases when `usecols` is an int or a list of ints I make the code read the columns indicated by `usecols` and after that I make `usecols` None, so nothing gets wrong in TextParser.

Reply:
This is pretty hairy, but I do see what you're trying to do to preserve compatibility.  Long term probably better off if we sync on usecols behavior with `read_csv`